### DISCUSSION TITLE
webcam driver installation in icicle-kit

### DISCUSSION THREAD

**Discussion by pratap635** [polarfire-soc/polarfire-soc-documentation#249] [How to] [answered]
I'm trying to install drivers and webcam application in icicle-kit but getting "command not found" error for most of the commands. Is it possible to install drivers and applications on board?

Can I interface webcam to USB OTG port of icicle-kit ? this USB OTG is used for image flash on eMMC memory.

**Comment by hughbreslin** [comment, marked as answer]
Hi @pratap635 

Please check out the [Working with PolarFire SoC USB Linux driver](https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/software-development/polarfire-soc-usb.md#working-with-polarfire-soc-usb-linux-driver) documentation for this, there is a dedicated section on working with webcams.

**Reply by pratap635** [reply]
For webcam interface we need to include packages in yocto image , how to include packages using windows 10
![Screenshot_2022-11-21-20-53-06-726_com android chrome](https://user-images.githubusercontent.com/92020114/203092433-2168be2a-b970-40f7-a954-78fed4674f9d.jpg)

**Reply by hughbreslin** [reply]
The best recommendation we have for doing a Linux build from a Windows host is using a Linux virtual machine. You could either transfer the Yocto .wic file to the Windows host and program from there or pass through the USB and program directly from the virtual machine. 

I don't think Yocto is fully supported by WSL (Windows Subsystem for Linux) and this isn't something we regularly test.

The best option for building Linux is from a Linux host PC.

**Reply by pratap635** [reply]
Can you please share the link how to include packages in yocto image.

**Reply by hughbreslin** [reply]
There are several examples available online, for example [here](https://variwiki.com/index.php?title=Adding_yocto_packages)
